src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java

```diff
@@ -153,8 +154,32 @@ public Map<String, ClinicalDataType> getClinicalAttributeDatatypeMap(StudyViewFi
     )
     @Override
     public List<ClinicalDataCountItem> getClinicalDataCounts(StudyViewFilter studyViewFilter, List<String> filteredAttributes) {
-       return studyViewRepository.getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes);
+
+        var result = studyViewRepository.getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes);
+        
+        // fetch the samples by using the provided study view filter
+        List<Sample> filteredSamples = getFilteredSamples(studyViewFilter);
+
+        var resultWithAllAttributes = StudyViewColumnarServiceUtil.addClinicalDataCountsForMissingAttributes(result, filteredAttributes, filteredSamples);
+
+        resultWithAllAttributes.stream().forEach(attr -> {
+            Map<String, List<ClinicalDataCount>> countsPerType = attr.getCounts().stream()
+                .collect(Collectors.groupingBy(ClinicalDataCount::getValue));
+            List<ClinicalDataCount> res = countsPerType.entrySet().stream().map((entry)->{
+                ClinicalDataCount mergedCount = new ClinicalDataCount();
+                mergedCount.setAttributeId(attr.getAttributeId());
+                mergedCount.setValue(entry.getKey());
+                mergedCount.setCount(entry.getValue().stream().mapToInt(ClinicalDataCount::getCount).sum());
+                return mergedCount;
+            }).collect(Collectors.toList());
+            attr.setCounts(res);
+        });
```

Comment:
Did you mean to use StudyViewColumnarServiceUtil.mergeClinicalDataCounts?

Reply:
good point 😆